### PR TITLE
Problem:(CRO-517) chain-abci doesn't send app_hash if there are no transactions

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution:  abci send apphash for empty blocks

### CODE REVIEW

**Comment by leejw51crypto** [comment]
thanks @tomtau

File: chain-abci/src/app/commit.rs [hunk 1]
```diff
@@ -43,7 +43,7 @@ impl<T: EnclaveProxy> ChainNodeApp<T> {
         let mut new_state = orig_state.expect("executing block commit, but no app state stored (i.e. no initchain or recovery was executed)");
         let mut resp = ResponseCommit::new();
         let mut inittx = self.storage.db.transaction();
-        if !self.delivered_txs.is_empty() {
+        {
```
Comment (calvinaco) [inline chain-abci/src/app/commit.rs]: Does it need a block scope?

File: chain-abci/src/app/commit.rs [hunk 2]
```diff
@@ -43,7 +43,7 @@ impl<T: EnclaveProxy> ChainNodeApp<T> {
         let mut new_state = orig_state.expect("executing block commit, but no app state stored (i.e. no initchain or recovery was executed)");
         let mut resp = ResponseCommit::new();
         let mut inittx = self.storage.db.transaction();
-        if !self.delivered_txs.is_empty() {
```
Comment (tomtau) [inline chain-abci/src/app/commit.rs]: I'd suggest to keep this `!self.delivered_txs.is_empty()` check, but to move out the steps  on L123 to L142

**Comment by tomtau** [comment]
bors try

File: chain-abci/src/app/commit.rs (hunk 2 above)
Comment (tomtau) [reply chain-abci/src/app/commit.rs]: or remove the block scope as @calvinaco suggested https://github.com/crypto-com/chain/pull/530#discussion_r339448866

File: chain-abci/src/app/commit.rs (hunk 1 above)
Comment (leejw51crypto) [reply chain-abci/src/app/commit.rs]: no, i'll remove it

File: chain-abci/src/app/commit.rs (hunk 2 above)
Comment (leejw51crypto) [reply chain-abci/src/app/commit.rs]: ok, i'll modify

**Comment by leejw51crypto** [comment]
tidy up

**Pull request merged**